pkg/lexer/lexer.go

```diff
@@ -69,6 +69,9 @@ func (l *Lexer) NextToken() token.Token {
 			if isTypePattern(tok.Literal) {
 				tok.Type = token.TYPE_PATTERN
 			}
+			if isLongOperator(tok.Literal) {
```

Comment:
should this be before isTypePattern?

Reply:
Nope. isLongOperator is much more strict than isTypePattern

Reply:
we need isLongOperator to take precedence over isTypePattern.  I misunderstood the code.  isLongOperator is taking precedence over isTypePattern when encountered.

Reply:
The idiomatic code for precedence should be:
```
			if isLongOperator(tok.Literal) {
				tok.Type = token.LookupOperator(tok.Literal)
			} else if isTypePattern(tok.Literal) {
				tok.Type = token.TYPE_PATTERN
			}
			return tok
```